src/mame/namco/namcos10.cpp

```diff
@@ -2607,25 +2764,76 @@ void namcos10_memp3_state::namcos10_memp3_base(machine_config &config)
 	m_maincpu->subdevice<psxdma_device>("dma")->install_write_handler(5, psxdma_device::write_delegate(&namcos10_memp3_state::pio_dma_read, this));
 	m_maincpu->set_addrmap(AS_PROGRAM, &namcos10_memp3_state::namcos10_memp3_map);
 
-	TMP95C061(config, m_memp3_mcu, XTAL(16'934'400));
+	TMP95C061(config, m_memp3_mcu, XTAL(101'491'200) / 2); // Measured
 	m_memp3_mcu->set_addrmap(AS_PROGRAM, &namcos10_memp3_state::mcu_map);
-	// Port 7 is used for communicating with the MP3 decoder chip
 
-	// LC82310 16.9344MHz
+	m_memp3_mcu->port1_read().set(FUNC(namcos10_memp3_state::port_read<1>));
+	m_memp3_mcu->port5_read().set(FUNC(namcos10_memp3_state::port_read<5>));
+	m_memp3_mcu->port7_read().set(FUNC(namcos10_memp3_state::port_read<7>));
+	m_memp3_mcu->port8_read().set(FUNC(namcos10_memp3_state::port_read<8>));
+	m_memp3_mcu->port9_read().set(FUNC(namcos10_memp3_state::port_read<9>));
+	m_memp3_mcu->porta_read().set(FUNC(namcos10_memp3_state::port_read<10>));
+	m_memp3_mcu->portb_read().set(FUNC(namcos10_memp3_state::port_read<11>));
+	m_memp3_mcu->port1_write().set(FUNC(namcos10_memp3_state::port_write<1>));
+	m_memp3_mcu->port2_write().set(FUNC(namcos10_memp3_state::port_write<2>));
+	m_memp3_mcu->port5_write().set(FUNC(namcos10_memp3_state::port_write<5>));
+	m_memp3_mcu->port6_write().set(FUNC(namcos10_memp3_state::port_write<6>));
+	m_memp3_mcu->port7_write().set(FUNC(namcos10_memp3_state::port_write<7>));
+	m_memp3_mcu->port8_write().set(FUNC(namcos10_memp3_state::port_write<8>));
+	m_memp3_mcu->porta_write().set(FUNC(namcos10_memp3_state::port_write<10>));
+	m_memp3_mcu->portb_write().set(FUNC(namcos10_memp3_state::port_write<11>));
+
+	m_memp3_mcu->an_read<0>().set([this] () {
+		return m_p3_analog[0].read_safe(0);
+	});
+
+	m_memp3_mcu->an_read<1>().set([this] () {
+		return m_p3_analog[1].read_safe(0);
+	});
+
+	m_memp3_mcu->an_read<2>().set([this] () {
+		return m_p3_analog[2].read_safe(0);
+	});
+
+	m_memp3_mcu->an_read<3>().set([this] () {
+		return m_p3_analog[3].read_safe(0);
+	});
+
+	LC82310(config, m_lc82310, XTAL(16'934'400));
+	m_lc82310->add_route(0, ":lspeaker", 1.0);
+	m_lc82310->add_route(1, ":rspeaker", 1.0);
```

Comment:
Do you really need absolute tags here?  Won’t `"lspeaker"` and `"rspeaker"` work?